src/lib/router.ts

```diff
@@ -21,10 +21,56 @@ router.post('/matches', async (req, res, next) => {
 
     const inserted = await getMatchesCollection().insertOne(match);
     if (!inserted.acknowledged) {
-      res.status(500).send('Error inserting match');
+      response.status(500).send('Error inserting match');
       return;
     }
-    res.status(200).json(match);
+    response.status(200).json(match);
+  } catch (error) {
+    next(error);
+  }
+});
+
+router.post('/matches/:gameId/highlights', async (request, response, next) => {
+  try {
+    const { timestamp, type, videoSrc } = request.body;
+    const { gameId } = request.params;
+    const requestisInvalid =
+      typeof timestamp !== 'number' ||
+      typeof type !== 'string' ||
+      typeof videoSrc !== 'string';
+
+    if (requestisInvalid) {
+      response.status(400).send('Invalid payload');
+      return;
+    }
+
+    const highlight: MatchHighlight = {
+      timestamp,
+      type,
+      videoSrc,
+    };
+
+    const inserted = await getMatchesCollection().updateOne(
+      { gameId: Number(gameId) },
+      { $push: { highlights: highlight } }
+    );
+
+    if (!inserted.matchedCount) {
+      response.status(404).send('Error pushing highlight: Match not found.');
+      return;
+    }
+    response.status(200).json(highlight);
+  } catch (error) {
+    next(error);
+  }
+});
+
+router.get('/matches', async (_request, response, next) => {
+  try {
+    const matchesCollection = await getMatchesCollection();
+    const matches = await matchesCollection.find({}).toArray();
```

Comment:
I like that you create a variable `matchesCollection`, but you didn't do that in the other handlers. I would be more consistent.